<launch>
  
  <arg name="model" default="$(find wpb_mani_description)/urdf/wpb_mani.urdf.xacro"/>
  <arg name="gui" default="false" />

  <param name="robot_description" command="$(find xacro)/xacro $(arg model)" />
  <param name="use_gui" value="$(arg gui)"/>
  <arg name="rvizconfig" default="$(find wpb_mani_tutorials)/rviz/slam.rviz" />
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

  <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping">
    <param name="base_frame" value="base_footprint"/>
  </node>

  <!-- wpb_mani core-->
  <node pkg="wpb_mani_bringup" type="wpb_mani_core" name="wpb_mani_core" output="screen">
    <param name="serial_port" type="string" value="/dev/ftdi"/> 
    <rosparam file="$(find wpb_mani_bringup)/config/wpb_mani.yaml" command="load" />
  </node>

  <!--- Run Rplidar -->
  <node name="rplidarNode"          pkg="rplidar_ros"  type="rplidarNode" output="screen">
    <param name="serial_port"         type="string" value="/dev/rplidar"/>  
    <param name="serial_baudrate"     type="int"    value="115200"/>
    <param name="frame_id"            type="string" value="laser"/>
    <param name="inverted"            type="bool"   value="false"/>
    <param name="angle_compensate"    type="bool"   value="true"/>
    <remap from="scan" to="scan_raw"/>
  </node>
    
  <!-- Run lidar filter -->
  <node pkg="wpb_mani_bringup" type="wpb_mani_lidar_filter" name="wpb_mani_lidar_filter">
    <param name="pub_topic" value="/scan"/>
  </node>

  <!-- joy node -->
  <node respawn="true" pkg="joy" type="joy_node" name="wpb_mani_joy" >
    <param name="dev" type="string" value="/dev/input/js0" />
    <param name="deadzone" value="0.12" />
  </node>

  <!-- Axes Velcmd -->
  <param name="axis_linear" value="1" type="int"/>
  <param name="axis_angular" value="0" type="int"/>
  <param name="scale_linear" value="1" type="double"/>
  <param name="scale_angular" value="1" type="double"/>
  <node pkg="wpb_mani_bringup" type="wpb_mani_js_vel" name="teleop"/>

</launch>